### PR TITLE
12 implement ipaymentservice iuploadservice

### LINKED ISSUE #12: Implement IPaymentService & IUploadService

### PR DESCRIPTION
**Pull request by Partvish**

### CODE REVIEW

**Review by DAbraham2 — CHANGES_REQUESTED** [review]
Add dependency for BlobClientService

**Review by DAbraham2 — APPROVED** [review]

File: src/DomainServices/Core/UploadService.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                        {
+                            await stream.CopyToAsync(fileStream);   
+                        }
+                    }
+                    return fileName;
+                }
+            }
+            catch(IOException e)
+            {
+               Console.WriteLine("Error occured during file save: " + e.Message);
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Change to `logger.LogError(e, "Error occured during file save);`

File: src/DomainServices/Core/UploadService.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        public async Task<bool> RemoveFileAsync(string fileLocation)
+        {
+            try
+            {
+                if (!this.options.ShouldUploadToAzure)
+                {
+                    if (File.Exists(fileLocation))
+                    {
+                        File.Delete(fileLocation);
+                        Console.WriteLine("File deleted");
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Change to `logger.LogInformation("File deleted");`

File: src/DomainServices/Core/UploadService.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                    }
+                    else
+                    {
+                        Console.WriteLine("File not found");
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Change to `logger.LogInformation("File not found");`

File: src/DomainServices/Core/UploadService.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                {
+                    BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(blobContainerName);
+                    BlockBlobClient blockBlobClient = containerClient.GetBlockBlobClient(fileLocation);
+                    blockBlobClient.DeleteIfExists();
+                }
+                return true;
+            }
+            catch (IOException e)
+            {
+                Console.WriteLine("Error occured during file deletion: " + e.Message);
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Change to `logger.LogError(e, "Error occured during file deletion");`

File: src/DomainServices/Core/UploadService.cs [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                        {
+                            stream.CopyTo(fileStream);
+                        }
+                    }
+                    return fileName;
+                }
+            }
+            catch (IOException e)
+            {
+                Console.WriteLine("Error occured during file save: " + e.Message);
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Change to `logger.LogError(e, "Error occured during file save");`

File: src/DomainServices/Service/PaymentService.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        private IDataService dataService;
+        public PaymentService(IDataService dataService)
+        {
+            this.dataService = dataService;
+        }
+        public async Task<bool> BuyItemAsync(int userId, int price, int caffId)
+        {
+            if (price < 0)
+                return false;
+            CaffDTO caffDTO = await this.dataService.AsyncGetCaff(caffId);
```
Comment (kerekesaron) [inline src/DomainServices/Service/PaymentService.cs]: Change to `GetCaffAsync`

**Review by kerekesaron — CHANGES_REQUESTED** [review]
Check comments

File: src/DomainServices/Core/UploadService.cs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+using Azure.Storage.Blobs.Specialized;
+
+namespace ShoppingLikeFiles.DomainServices.Core
+{
+    class UploadService: IUploadService
+    {
+        private UploadServiceOptions options;
+        private BlobServiceClient blobServiceClient;
+        private string blobContainerName="caff_container";
+        public UploadService(UploadServiceOptions options, string connectionString)
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Inject `ILogger<UploadService> logger`

**Review by kerekesaron — CHANGES_REQUESTED** [review]

File: src/DomainServices/Core/UploadService.cs [hunk 8]
```diff
@@ -15,8 +16,10 @@ class UploadService: IUploadService
         private UploadServiceOptions options;
         private BlobServiceClient blobServiceClient;
         private string blobContainerName="caff_container";
-        public UploadService(UploadServiceOptions options, string connectionString)
+        private ILogger logger;
+        public UploadService(ILogger logger, UploadServiceOptions options, string connectionString)
```
Comment (kerekesaron) [inline src/DomainServices/Core/UploadService.cs]: Az ILogger vegye át template paraméterként az osztályt, ahogy korábban is kértem:
`ILogger<UploadService>`

**Review by kerekesaron — CHANGES_REQUESTED** [review]

**Review by kerekesaron — APPROVED** [review]

**Pull request merged**